### PR TITLE
Add home controller test

### PR DESCRIPTION
**Pull request by yksflip**
Co-authored-by: viehlieb <pf@pragma-shift.net>
Co-authored-by: Tobias Kneuker <tk@pragma-shift.net>

### CODE REVIEW

**Comment by wvengen** [comment]
At first glance, great! I can imagine this is useful especially for the Rails upgrade. :+1:

**Comment by yksflip** [comment]
thanks wvengen! 
@paroga  what do you think? Anything needed to get this merged? :)

File: spec/controllers/home_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  describe 'GET index' do
+    describe 'NOT logged in' do
+      it 'redirects' do
+        get_with_defaults :profile
+        expect(response).to have_http_status(:redirect)
+        expect(response).to redirect_to(login_path)
+      end
+    end
+
+    describe 'logegd in' do
```
Comment (lentschi) [inline spec/controllers/home_controller_spec.rb:17]: typo:
```suggestion
    describe 'logged in' do
```

File: Gemfile [hunk 2]
```diff
@@ -112,6 +112,7 @@ group :test do
   gem 'rspec-core'
   gem 'rspec-rerun'
   gem 'i18n-spec'
+  gem 'rails-controller-testing'
```
Comment (lentschi) [inline Gemfile:115]: Honestly, considering what is being said in the linked [deprecation message](https://github.com/rails/rails/issues/18950#issue-57746192), I'm not sure, if it's a good idea to use this gem at all:
> Testing what instance variables are set by your controller is a bad idea. 
> [...]The same goes for assert_template.

But I don't feel very strongly about it, so if you really want to use it and noone else has objections, I'm fine with it!

File: spec/controllers/home_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+      before { login user }
+
+      it 'assigns tasks' do
+        get_with_defaults :index
+
+        expect(assigns(:unaccepted_tasks)).not_to be_nil
+        expect(assigns(:next_tasks)).not_to be_nil
+        expect(assigns(:unassigned_tasks)).not_to be_nil
+        expect(response).to render_template('home/index')
```
Comment (lentschi) [inline spec/controllers/home_controller_spec.rb:26]: See my comment about the `rails-controller-testing` gem above. But if this gem _is_ going to be used after all, couldn't you use this instead?:

```suggestion
        assert_template 'home/index'
```

File: app/controllers/home_controller.rb [hunk 4]
```diff
@@ -18,15 +18,15 @@ def reference_calculator
       @bank_accounts = @types.includes(:bank_account).map(&:bank_account).uniq.compact
       @bank_accounts = [BankAccount.last] if @bank_accounts.empty?
     else
-      redirect_to root_url, alert: I18n.t('group_orders.errors.no_member')
+      redirect_to root_path, alert: I18n.t('group_orders.errors.no_member')
```
Comment (lentschi) [inline app/controllers/home_controller.rb:21]: The PR says this is about adding a test, not about changing redirect behavior. Has this maybe slipped in from some other branch? (Other than that I'm fine with it!)

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -21,6 +21,10 @@
 
 RSpec.configure do |config|
   # We use capybara with webkit, and need database_cleaner
+  config.before(:suite) do
+    DatabaseCleaner.clean_with(:truncation)
+  end
```
Comment (lentschi) [inline spec/spec_helper.rb:26]: Just out of curiosity: Why did you add this? (Tests seem to run fine without it. Could it be this has accidentally slipped in from some other change?)

**Review by lentschi — COMMENTED** [review]
Looks fine to me in general; added some minor remarks

File: app/controllers/home_controller.rb (hunk 4 above)
Comment (yksflip) [reply app/controllers/home_controller.rb:21]: I tried to make the usage of  _path and _url uniform, but no need to do it here...
The change in l67 is actually a fix see https://github.com/foodcoops/foodsoft/pull/970#discussion_r1039455645

File: Gemfile (hunk 2 above)
Comment (yksflip) [reply Gemfile]: I think you are right, we should try to conform with the rails conventions.
I'll rework the controller tests and remove tests that step out of the boundaries and try to add the parts as integration tests.

File: Gemfile.lock [hunk 6]
```diff
@@ -642,4 +642,4 @@ DEPENDENCIES
   whenever
 
 BUNDLED WITH
-   1.17.3
+   2.4.5
```
Comment (lentschi) [inline Gemfile.lock:645]: I think, since there are now no changes to the `Gemfile` left, `Gemfile.lock` should remain untouched too. right?

**Review by lentschi — CHANGES_REQUESTED** [review]

File: Gemfile.lock (hunk 6 above)
Comment (yksflip) [reply Gemfile.lock]: yess you're right!

**Comment by wvengen** [comment]
Hooray! :tada:
